vmdb/app/models/ems_microsoft/powershell.rb

```diff
@@ -38,6 +38,10 @@ def log_dos_error_results(results)
         log_header = "MIQ(#{self.class.name}##{__method__})"
         error = results[:data].collect { |d| d[:stderr] }.join
         $scvmm_log.error("#{log_header} #{error}") unless error.blank?
+
+        if error.downcase.include? "'get-scvmmserver' is not recognized"
+          raise "Hyper-V Server is not directly supported. "
+        end
```

Comment:
This line `error = results[:data].collect { |d| d[:stderr] }.join` feels like it should be its own method named something like `extract_error_from_results`.  Then you just do that immediately after getting results.  I don't think it belongs in a log_* method.